### PR TITLE
Using prctl(PR_SET_MM, PR_SET_MM_MAP,...) on Linux instead of argv clobbering

### PR DESCRIPTION
**Pull request by gershnik**
This solves numerous issues such as dependence on non-modified environ, existence of Py_GetArgcArgv (and getting argv at all), bug #51, etc.

The downside is that it is hard to tell whether it will work on all reasonable Linux installations. The docs for `PR_SET_MM` say that it requires `CAP_SYS_RESOURCE` capability. My experiments with Ubuntu 20.04 seem to indicate that it isn't so - it works just fine with capability removed. Perhaps it is different on SELinux but I don't have one handy to check.  (OTOH inability to change title without extra steps on SELinux might be an upside...). In addition this also bumps minimum kernel requirement to 3.18 I believe. 

It is likely possible to check whether PR_SET_MM works at runtime and fall back on argc/argv but that's a whole ton of additional coding work.

In short, I am not 100% sure about this one so take it or leave it as you see fit 😀

### CODE REVIEW

**Comment by solardiz** [comment]
Beware: looks like the code in this PR will misbehave when the old process name contains space character(s) in it. See these threads:
https://twitter.com/dvyukov/status/1605507498420473857
https://www.openwall.com/lists/oss-security/2022/12/21/6

**Comment by gershnik** [comment]
Yeah, the code everybody is using to parse it is broken. Linux insistence on communicating system information via human readable strings rather than an API bites again. I no longer have time or interest to try to fix it so closing.
